dsync/fetch.go

```diff
@@ -143,6 +176,9 @@ func (f *Fetch) Do() (err error) {
 							f.prog[i] = 100
 						}
 					}
+					// why does this happen in a go process?
```

Comment:
Don't ask questions in comments. File an issue instead.

To answer the question itself, `fetch.completionChanged()` will block until something reads from the internal `progCh`, which nothing is currently doing, so without the `go` in front this would block forever. The long-term use for progCh is to allow callers a way to receiving updates while the sync process is executing. Callers may want this to show a progress bar, or bail if time between block-transmission is too long.